asn1/Stream.asn

```diff
@@ -0,0 +1,177 @@
+Stream
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt96,
+    UInt128,
+    VarUInt,
+    VarBytes
+    FROM GenericTypes
+
+    Address
+    FROM InterledgerTypes
+;
+
+StreamEncryptionEnvelope ::= SEQUENCE {
+    -- Random nonce
+    initializationVector UInt96,
+
+    -- Authentication tag (output of AES-GCM encryption)
+    authenticationTag UInt128,
+
+    -- Encrypted STREAM Packet
+    -- Note: This is NOT encoded as a variable length (length-prefixed)
+    -- octet string. The size is constrained by the length of the data
+    -- field in the InterledgerPrepare, InterledgerFulfill, or InterledgerReject
+    -- packet containing this StreamEncryptionEnvelope
+    cipherText OCTET STRING (SIZE (0..32739))
+}
+
+StreamPacket ::= SEQUENCE {
+  -- Always 1 for now
+  version UInt8,
+
+  -- ILP Packet type this STREAM packet must be carried upon
+  ilpPacketType UInt8,
+
+  -- Packet ID
+  sequence VarUInt,
```

Comment:
Since we recommend closing connections after 2^32 packets have been sent couldn't this be fixed length?

Reply:
Yeh, I think if you made this UInt32 it would still be less efficient than VarUInt for the majority of packets.